projects/control-service/projects/pipelines_control_service/src/main/java/com/vmware/taurus/service/deploy/JobImageBuilder.java

```diff
@@ -253,14 +256,26 @@ public boolean isBuildingJobInProgress(String dataJobName) {
   private Map<String, String> getBuildParameters(DataJob dataJob, JobDeployment jobDeployment) {
     String jobName = dataJob.getName();
     String jobVersion = jobDeployment.getGitCommitSha();
+    String baseImage = null;
+    String pythonVersion = jobDeployment.getPythonVersion();
+
+    // TODO: Remove this part when datajobs.deployment.dataJobBaseImage is deprecated.
+    if (deploymentDataJobBaseImage != null) {
+      baseImage = deploymentDataJobBaseImage;
+    } else if (pythonVersion != null
+        && supportedPythonVersions.isPythonVersionSupported(pythonVersion)) {
+      baseImage = supportedPythonVersions.getJobBaseImage(pythonVersion);
+    } else {
+      baseImage = supportedPythonVersions.getDefaultJobBaseImage();
+    }
```

Comment:
```suggestion
    baseImage = deploymentDataJobBaseImage != null ? deploymentDataJobBaseImage : supportedPythonVersions.getJobBaseImage(pythonVersion);
```

Isn't it the same?

Reply:
I think the @ivakoleva suggestion is better. In other words, incorporate this logic in `supportedPythonVersions.getJobBaseImage()`.

Reply:
Moved to SupportedPythonVersions. Thank you for the review